ports/azure-messaging-eventhubs-cpp/portfile.cmake

```
@@ -5,8 +5,10 @@ vcpkg_from_github(
     SHA512 3677edf50218c39e378a968661c0ec0600b8760d3af33f204f3267e104449c0aef42b9520cb63b5e5810b89bf91f06979d009ffdd6d6d04d2c14e1a21f0e111f
 )
 
+file(RENAME ${SOURCE_PATH}/sdk/eventhubs/azure-messaging-eventhubs/ ${SOURCE_PATH}/sdk/eventhubs/eventhubs/)

```

```suggestion
if(EXISTS "${SOURCE_PATH}/sdk/eventhubs/azure-messaging-eventhubs")
    file(REMOVE_RECURSE "${SOURCE_PATH}/sdk/eventhubs/eventhubs")
    file(RENAME "${SOURCE_PATH}/sdk/eventhubs/azure-messaging-eventhubs" "${SOURCE_PATH}/sdk/eventhubs/eventhubs")
endif()
```
The `if` guard is for `--editable` installations.

---

I understand that you are in a hurry to unblock *your CI*.
But I suggested more radical shortening (than just 16 chars) in order to help *your users*. Which may also be unaware of `--x-buildtrees-root`.

```suggestion
if(EXISTS "${SOURCE_PATH}/sdk/eventhubs/azure-messaging-eventhubs")
    file(REMOVE_RECURSE "${SOURCE_PATH}.renamed")
    file(RENAME ${SOURCE_PATH}/sdk/eventhubs/azure-messaging-eventhubs" "${SOURCE_PATH}.renamed")
    file(REMOVE_RECURSE "${SOURCE_PATH}")
    file(RENAME ${SOURCE_PATH}.renamed" "${SOURCE_PATH}")
endif()
```
assuming that nothing outside `azure-messaging-eventhubs` is needed.

---

Thanks! I'll add the `if` guard, but we don't want to shorten the number of levels in directories - on one occasion, library's cmake does include a sibling header-only library as `../azure-security-keyvault-shared`, and all our libraries do include cmake scripts via `../../../cmake-modules`. Therefore, as of radical renames, we are thinking of renaming `sdk/service/library/` (for the port being installed) to `_/_/_/` - https://github.com/Azure/azure-sdk-for-cpp/pull/4977. This is the most radical we can go at the moment.